File: policyliner/src/main/java/de/tub/dima/policyliner/dto/QueryRequestDTO.java
package de.tub.dima.policyliner.dto;

import de.tub.dima.policyliner.constants.QueryComparatorType;

public class QueryRequestDTO {
    private String query;
    private String userId;
    private String userRole;
    private QueryComparatorType comparatorType;


    public QueryRequestDTO(String query, String userId, String userRole, QueryComparatorType comparatorType) {
        this.query = query;
        this.userId = userId;
        this.userRole = userRole;
        this.comparatorType = comparatorType;
    }

    public String getQuery() {
        return query;
    }

    public void setQuery(String query) {
        this.query = query;
    }

    public String getUserId() {
        return userId;
    }

    public void setUserId(String userId) {
        this.userId = userId;
    }

    public String getUserRole() {
        return userRole;
    }

    public void setUserRole(String userRole) {
        this.userRole = userRole;
    }

    public QueryComparatorType getComparatorType() {
        return comparatorType;
    }

    public void setComparatorType(QueryComparatorType comparatorType) {
        this.comparatorType = comparatorType;
    }
}

File: policyliner-demo-ui/src/app/dtos/queryRequest.dto.ts
import {QueryComparatorType} from './queryComparatorType.enum';

export class QueryRequestDTO {
  query?: string;
  userId?: string;
  userRole?: string;
  comparatorType?: QueryComparatorType;

  constructor(obj?: any) {
    if (obj) {
      this.query = obj.query;
      this.userId = obj.userId;
      this.userRole = obj.userRole;
      this.comparatorType = obj.comparatorType;
    }
  }

}
